docs/user/admin/settings.rst

```diff
@@ -188,6 +188,51 @@ Result set::
       }
     }
 
+plugins.query.system_limit
+==========================
+
+Description
+-----------
+
+The size configures the maximum amount of documents to be pull from OpenSearch for data-intensive operations (e.g. join, lookup). The default value is: 50000.
```

Comment:
The PR description provides a clearer definition. The size parameter specifies the maximum allowable data bloat from input documents during operations.